services/types/couchapp/lib/schemas/generic.js

```diff
@@ -3,19 +3,25 @@ exports.v1 = {
   "$schema": "http://json-schema.org/draft-04/schema#",
   "version": "1.0",
   "type": "object",
-  "properties": {
-    "_id": {
+  "description": "this schema provide the overall document structure",
+  "definitions": {
+    "alphanumeric": {
       "type": "string",
       "pattern": "^[0-9A-Za-z-_]*$"
     },
-    "type": {
+    "url":{
       "type": "string",
-      "pattern": "^[0-9A-Za-z-_]*$"
+      "pattern": "(http|https):\\/\\/(\\w+:{0,1}\\w*@)?(\\S+)(:[0-9]+)?(\\/|\\/([\\w#!:.?+=&%@!\\-\\/]))?"
```

Comment:
Good remark. It doesn't apply here imho because we don't know if the producer serve content on both http and https so it's safer to ask explicitly